chips/sam4l/src/usbc/register_macros.rs

```diff
@@ -0,0 +1,65 @@
+//! Macros for defining USBC registers
```

Comment:
These are wonderful macros! Most the comments I would have here are really about my comments in the common_register module.